### PR TITLE
PolicyBot improvements

### PR DESCRIPTION
**Pull request by geeknoid**
- Only run the lifecycle update logic for a changed issue is the change was
performed by an org member. This should prevent the bot fighting with itself
and causing multiple comments being added to the same issue.

- Add missing logic to remove the stale label & comment from issues that don't
need them. They were being left behind.

### CODE REVIEW

File: policybot/mgrs/lifecyclemgr/mgr.go [hunk 1]
```diff
@@ -71,6 +75,11 @@ func (lm *LifecycleMgr) ManageAll(context context.Context) error {
 			}
 
 			for _, issue := range issues {
+
+				if issue.IssueNumber != 1101 {
```
Comment (howardjohn) [inline policybot/mgrs/lifecyclemgr/mgr.go]: remove?

File: policybot/mgrs/lifecyclemgr/mgr.go [hunk 2]
```diff
@@ -62,6 +62,10 @@ func (lm *LifecycleMgr) ManageAll(context context.Context) error {
 		for _, r := range o.Repos {
 			var st Stats
 
+			if r.Name != "api" {
+				continue
```
Comment (howardjohn) [inline policybot/mgrs/lifecyclemgr/mgr.go]: was this for debugging?

File: policybot/mgrs/lifecyclemgr/mgr.go (hunk 1 above)
Comment (geeknoid) [reply policybot/mgrs/lifecyclemgr/mgr.go]: Ouch, yes, this was for debgging.

File: policybot/mgrs/lifecyclemgr/mgr.go (hunk 2 above)
Comment (geeknoid) [reply policybot/mgrs/lifecyclemgr/mgr.go]: Yep. Thanks.

**Pull request merged**